pkg/apis/machine/v1beta1/machine_webhook.go

```diff
@@ -612,6 +640,28 @@ func validateAWS(m *Machine, clusterID string) (bool, utilerrors.Aggregate) {
 				"expected providerSpec.credentialsSecret to be populated",
 			),
 		)
+	} else {
+		secretExists, err := secretExists(*c, providerSpec.CredentialsSecret.Name, m.GetNamespace())
+		if err != nil {
+			errs = append(
+				errs,
+				field.Invalid(
+					field.NewPath("providerSpec", "credentialsSecret"),
+					providerSpec.CredentialsSecret.Name,
+					fmt.Sprintf("failed to get credentialsSecret: %v", err),
+				),
+			)
+		}
+		if !secretExists && err == nil {
+			errs = append(
+				errs,
+				field.Invalid(
+					field.NewPath("providerSpec", "credentialsSecret"),
+					providerSpec.CredentialsSecret.Name,
+					fmt.Sprintf("not found. Expected CredentialsSecret to exist"),
+				),
+			)
+		}
```

Comment:
Is this identical for each provider? Could we extract it to a helper function maybe?

I'm thinking something like `errs = append(errs, credentialsSecretExists(args)...)`